pkg/client/record/event.go

```diff
@@ -25,11 +26,23 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
 
 	"github.com/golang/glog"
 )
 
+const (
+	maxQueuedEvents  = 1000
+	maxTriesPerEvent = 10
+)
+
+var (
+	minSleep   = float64(1 * time.Second)
+	maxSleep   = float64(15 * time.Second)
+	backoffExp = 1.5
```

Comment:
Yeah, I'd like very easy-to-understand, predictable behavior here.

Suppose a master goes down for some period, and then comes back up. I'll posit that we should be able to reason about how many events the kubelets will have stored up, and what period of time they will deliver them to the master, and in what manner they will be delivered.

If the rule is "one wait randomly between 0 and N seconds, followed by M N second waits", then I think that when master comes back up, events ought to be delivered to it uniformly over N seconds.

Other rules, like this exponential backoff rule, don't let me predict the distribution or time period, at least not easily. I'm not even sure about the behavior of "N waits of rand(0, 5) seconds"-- I think, but am not sure, that this might cause event requests to be normally distributed instead of uniformly distributed, which would cause a much larger (though shorter) QPS spike.

Even if the above is overly paranoid, I'd still like the aggregate behavior of many clients releasing queued events to be very, very easy to reason about.

Exponential backoff is great for timing network packets to eventually miss each other. I don't actually think it's good for much else.